File: mobile/lib/config/server_config.dart
/// Central server configuration for Synchronization.
///
/// The app connects to the Cloudflare signaling proxy, which automatically
/// routes to the GCP VM (primary) or Render free tier (fallback).
/// Nothing else in the codebase should hardcode a server URL.
class ServerConfig {
  /// Cloudflare signaling proxy — smart router between VM and Render fallback.
  static const String primaryServer = 'https://sync.synchronizationpro.app';

  /// The active signaling server URL used by the entire app.
  static String get signalingServer => primaryServer;

  /// The active server host (without scheme/port) used for deep link validation.
  static String get signalingHost => Uri.parse(signalingServer).host;
}

File: extension/src/serverConfig.ts
/**
 * Central server configuration for the Synchronization extension.
 *
 * The extension connects to the Cloudflare signaling proxy, which
 * automatically routes to the GCP VM (primary) or Render (fallback).
 * Nothing else in the extension should hardcode a server URL.
 */

/** Cloudflare signaling proxy — smart router between VM and Render fallback. */
export const PRIMARY_SERVER = 'https://sync.synchronizationpro.app';

/** The active signaling server URL used by the entire extension. */
export const SIGNALING_SERVER = PRIMARY_SERVER;

/** The connect page base URL (for QR code generation). */
export const CONNECT_PAGE_URL = 'https://synchronizationpro.app/c';
